content/client-lib-development-guide/features.textile

```diff
@@ -118,7 +118,9 @@ h3(#rest-auth). Auth
 ** @(RSA3c)@ For Realtime websocket connections, the querystring param @accessToken@ is appended to the URL endpoint
 ** @(RSA3d)@ A test must exist that each type of token string is correctly passed in requests (ie according to @(RSA3b)@ and @(RSA3c)@)
 * @(RSA4)@ Token Auth is used if @useTokenAuth@ is set to true, or if @useTokenAuth@ is unspecified and any one of @authUrl@, @authCallback@, @token@, or @TokenDetails@ is provided
-** @(RSA4a)@ When a @token@ or @tokenDetails@ is used to instance the library, and no means to renew the token is provided (either an API key, @authCallback@ or @authUrl@), if the server responds with a token error (401 HTTP status code and an Ably error value @40140 <= code < 40150@), then the client library should indicate an error, not retry the request and in the case of the realtime library, transition the connection to the @FAILED@ state
+** @(RSA4a)@ When a @token@ or @tokenDetails@ is used to instance the library, and no means to renew the token is provided (either an API key, @authCallback@ or @authUrl@):
+*** @(RSA4a1)@ if the server responds with a token error (401 HTTP status code and an Ably error value @40140 <= code < 40150@), then the client library should indicate an error, not retry the request and in the case of the realtime library, transition the connection to the @FAILED@ state
+*** @(RSA4a2)@ a message at @info@ log level should be logged indicating that no means has been provided to renew the supplied token, with a link to a "solution article":https://help.ably.io/a_suitable_article with configuration advice relating to token authentication. 
```

Comment:
> with a link to a solution article

help.ably.io currently works on error codes, so we should probably define a code for this (we can use the same one for rsa4a1 and rsa4a2, both'd be directed to the same article about renewability)

cf https://github.com/ably/wiki/issues/142

Reply:
Yes, agreed, we just need to add a new error code.

Reply:
I was sure we had added this but now I don't see it.

Reply:
@paddybyers any progress on an error code?  Once done, @tomczoink please add a suitable solution article, or link to one if it exists (it should)